transports/websocket-websys/src/lib.rs

```diff
@@ -134,14 +135,26 @@ fn extract_websocket_url(addr: &Multiaddr) -> Option<String> {
         | (Some(Protocol::Dns4(h)), Some(Protocol::Tcp(port)))
         | (Some(Protocol::Dns6(h)), Some(Protocol::Tcp(port)))
         | (Some(Protocol::Dnsaddr(h)), Some(Protocol::Tcp(port))) => {
+            domain_found = true;
             format!("{}:{}", &h, port)
         }
         _ => return None,
     };
 
-    let (scheme, wspath) = match protocols.next() {
-        Some(Protocol::Ws(path)) => ("ws", path.into_owned()),
-        Some(Protocol::Wss(path)) => ("wss", path.into_owned()),
+    let (scheme, wspath) = match (protocols.next(), protocols.next()) {
+        (Some(Protocol::Tls), Some(Protocol::Ws(path))) => {
+            if !domain_found {
+                return None;
+            }
+            ("wss", path.into_owned())
+        }
+        (Some(Protocol::Ws(path)), _) => ("ws", path.into_owned()),
+        (Some(Protocol::Wss(path)), _) => {
+            if !domain_found {
+                return None;
+            }
+            ("wss", path.into_owned())
+        }
```

Comment:
I might be misreading, but wouldnt using `domain_found` to check for domains break previous impl, in other words, be a breaking change? Though im sure it might be rare (and would just be an additional san entry iirc), but there could be instances where one would have a certificate for a specific ip, no? If thats the route we are to go, it might be best to do a minor bump instead of a patch bump to the version.

Reply:
I did it to make it consistent with `libp2p-websocket` crate, however I reverted this check and I will also remove it from `libp2p-websocket` crate. It is valid to have an ip instead of a domain. For example:

```rust
use rustls::{ClientConfig, ClientConnection, RootCertStore, Stream};
use std::io::{Read, Write};
use std::net::TcpStream;
use std::sync::Arc;
use webpki_roots::TLS_SERVER_ROOTS;

fn main() {
    let provider = rustls::crypto::ring::default_provider();
    let cert_store = TLS_SERVER_ROOTS.iter().cloned().collect::<RootCertStore>();

    let client = ClientConfig::builder_with_provider(provider.into())
        .with_safe_default_protocol_versions()
        .unwrap()
        .with_root_certificates(cert_store)
        .with_no_client_auth();

    // 1.1.1.1 has valid TLS certificate
    let server_name = "1.1.1.1".try_into().unwrap();
    let mut conn = ClientConnection::new(Arc::new(client), server_name).unwrap();
    let mut sock = TcpStream::connect("1.1.1.1:443").unwrap();

    let mut io = Stream::new(&mut conn, &mut sock);

    io.write_all(b"GET / HTTP/1.1\r\n\r\n").unwrap();

    let mut resp = Vec::new();
    io.read_to_end(&mut resp).unwrap();

    println!("{}", String::from_utf8(resp).unwrap());
}
```

Reply:
Opened a separate PR for IP issue: https://github.com/libp2p/rust-libp2p/pull/5525